soh/src/code/z_parameter.c

```diff
@@ -2305,6 +2305,8 @@ u8 Item_Give(PlayState* play, u8 item) {
                         }
                     } else {
                         gSaveContext.equips.buttonItems[i] = ITEM_NONE;
+                        // No need to call OnItemGive for ITEM_SOLD_OUT. Doesn't have any ItemTable entries anyway.
+                        return -1;
```

Comment:
You're returning -1 here, but the return type of the function is a u8. Not sure if this number or the return type needs to change but something does.

Reply:
The return value of ItemGive is never used. I wouldn't know what else to make it do anyway.

Reply:
I'm not even sure why it has a return type to begin with.